### PR TITLE
AUTH_GET_PERMISSIONS

### PR DESCRIPTION
**Pull request by fonzarely**
This PR provides AUTH_GET_PERMISSIONS to authClient.
Permissions are saved in locaStorage, I'm not sure it's relevant but I'm expecting to prevent unnecessary jwt decode.
Permissions are decoded in AUTH_GET_PERMISSIONS and not in AUTH_LOGIN due to the fact the token may be provided via oauth so AUTH_LOGIN is not executed.

### CODE REVIEW

File: src/authClient.js [hunk 1]
```diff
@@ -23,17 +29,44 @@ export default (client, options = {}) => (type, params) => {
...
       }
-      return Promise.resolve();
+      return Promise.resolve()
+    case AUTH_GET_PERMISSIONS:
+      /*
+      JWT token may be providen by oauth, 
+      so that's why the permissions are decoded here and not in AUTH_LOGIN.
+      */
+      //Get the permissions from localstorage if any.
+      const permissions = JSON.parse(localStorage.getItem(permissionsKey));
```
Comment (kfern) [inline src/authClient.js]: localStorage can be manipulated. See https://stackoverflow.com/questions/9404813/how-to-view-or-edit-localstorage#comment78227724_12799050

**Comment by kfern** [comment]
Could not a service be consulted?

**Comment by josx** [comment]
Hey, but anyways those permission need to be done on also on backend

**Comment by kfern** [comment]
I think it is better to query a service instead of managing the permissions on the jwt

**Comment by josx** [comment]
can we make a option on how to handle it? i will glad to accept a PR on this.

File: src/authClient.js (hunk 1 above)
Comment (fonzarely) [reply src/authClient.js]: Yes it is true, here we are using permissions to adapt the display to user's permissions. You have to manage permissions too on the api. Here is a sample hook for that:
```javascript
restrictToRoles({
    roles: [
      'admin',
    ],
    fieldName: 'roles',
  })
```

**Comment by fonzarely** [comment]
@josx I see that my PR tests are failing and you had to fix this. Sorry for that.
Could you activate circle CI again?

**Comment by josx** [comment]
I dont understand why it is not working on PRs. 
Because i have actived from circleci options to do it. Now i have changed something.

**Comment by fonzarely** [comment]
I'll let you know for my next PR

**Comment by josx** [comment]
great!

**Pull request merged**